### PR TITLE
Add options chain snapshot

### PR DESCRIPTION
**Pull request by joedursun**
This PR adds support for the new options chain snapshot endpoint.

### CODE REVIEW

File: src/rest/options/snapshots.ts [hunk 1]
```diff
@@ -68,6 +68,41 @@ export interface IOptionsSnapshotContract {
...
+  contract_type?: string;
+  expiration_date?: string;
+  "expiration_date.lt"?: string;
+  "expiration_date.lte"?: string;
+  "expiration_date.gt"?: string;
+  "expiration_date.gte"?: string;
+  order?: "asc" | "desc";
+  limit?: number;
+  sort?: string;
+  strike_price?: number;
```
Comment (teambailey) [inline src/rest/options/snapshots.ts:90]: Should the `strike_price` filter parameters be added as well? ie. `"strike_price.lt"?: number`

File: src/rest/options/snapshots.ts [hunk 2]
```diff
@@ -68,6 +68,41 @@ export interface IOptionsSnapshotContract {
...
+export interface IOptionsSnapshotChain {
+  status?: string;
+  request_id?: string;
+  previous_url?: string;
+  next_url?: string;
+  results?: SnapshotInfo[];
+}
+
+export interface IOptionsChainQuery extends IPolygonQuery {
+  contract_type?: string;
```
Comment (teambailey) [inline src/rest/options/snapshots.ts:81]: Believe the accepted options should be `"call" | "put"` for `contract_type`

**Review by teambailey — CHANGES_REQUESTED** [review]

**Review by tharnach — COMMENTED** [review]
Should this PR should account for `strike_price` filters as well?

```
strike_price.lt - less than
strike_price.lte - less than or equal to
strike_price.gt - greater than
strike_price.gte - greater than or equal to

File: src/rest/options/snapshots.ts [hunk 3]
```diff
@@ -68,6 +68,41 @@ export interface IOptionsSnapshotContract {
...
+  expiration_date?: string;
+  "expiration_date.lt"?: string;
+  "expiration_date.lte"?: string;
+  "expiration_date.gt"?: string;
+  "expiration_date.gte"?: string;
+  order?: "asc" | "desc";
+  limit?: number;
+  sort?: string;
```
Comment (teambailey) [inline src/rest/options/snapshots.ts:89]: The accepted `sort` options should be `"ticker" | "expiration_date" | "strike_price"`

File: src/rest/options/snapshots.ts (hunk 1 above)
Comment (joedursun) [reply src/rest/options/snapshots.ts:90]: Yes good call!

**Review by teambailey — APPROVED** [review]

**Review by tharnach — APPROVED** [review]

**Review by mpatel18 — APPROVED** [review]
👍🏽

**Comment by jbonzo** [comment]
We are good to merge this, the API has been released

**Pull request merged**